elementary/monitor/api/tests/tests.py

```diff
@@ -267,20 +267,28 @@ def _get_invocations(
         self, test_result_db_rows: List[TestResultDBRowSchema]
     ) -> Dict[TestUniqueIdType, InvocationsSchema]:
         grouped_invocations = defaultdict(list)
+        grouped_invocation_ids = defaultdict(list)
         for test_result_db_row in test_result_db_rows:
             try:
                 elementary_unique_id = test_result_db_row.elementary_unique_id
-                grouped_invocations[elementary_unique_id].append(
-                    InvocationSchema(
-                        id=test_result_db_row.invocation_id
-                        or test_result_db_row.test_execution_id,
-                        time_utc=test_result_db_row.detected_at,
-                        status=test_result_db_row.status,
-                        affected_rows=self._parse_affected_row(
-                            results_description=test_result_db_row.test_results_description
-                        ),
-                    )
+                invocation_id = (
+                    test_result_db_row.invocation_id
+                    or test_result_db_row.test_execution_id
                 )
+                # Currently the way we flat test results causing that there is suplication in test incovation for each test.
+                # This if statement checks if the invocation is already counted or not.
+                if invocation_id not in grouped_invocation_ids[elementary_unique_id]:
+                    grouped_invocation_ids[elementary_unique_id].append(invocation_id)
```

Comment:
I don't understand why is there both `grouped_invocations` and `grouped_invocation_ids`.

Reply:
Invocation ids is the list of the invocation ids we found for the test.
I use it as an helper list to understand if I already added the same invocation to the grouped invocations object for the relevant test 🙂 
(I find it easier to look for an id inside a list of ids, instead of looking for id inside a list of objects that has the key id).

If you think that this is not simpler, I don't mind changing it 🙂

Reply:
I see, makes sense.